### ISSUE TITLE
ENH: IMM order selection uses ad-hoc formula — replace with BIC or AIC for principled model selection

### ISSUE THREAD

**Issue by roeimed0** [roeimed0/bacterial-gene-prediction#100]
## Problem

\`build_all_scoring_models()\` estimates the IMM order using:

\`\`\`python
estimated_order = math.floor(math.log2(effective_n / min_observations) / 2)
estimated_order = min(estimated_order, 8)
estimated_order = max(estimated_order, 3)
\`\`\`

This formula has no theoretical basis and no published reference. The hard clamps to [3, 8] are also arbitrary.

**Consequences:**
- For small genomes (Archaea, endosymbionts, 1–2 Mb), the formula may overestimate order, causing sparse k-mer counts and poor probability estimates
- For large genomes (>8 Mb), the formula is capped at 8 even if data supports higher order

## Biologically Correct Alternative

Use **BIC** (Bayesian Information Criterion) to select the order that best balances model fit vs complexity:

\`\`\`python
# For each candidate order k in [1..max_k]:
#   train IMM, compute log-likelihood on held-out training sequences
#   BIC(k) = -2 * log_likelihood + k * log(n_observations)
# Choose k* = argmin BIC(k)
\`\`\`

Alternatively, use a held-out validation set of training sequences (10% of training data) and select the order with the best held-out log-likelihood.

## Expected Impact

Better-calibrated IMM order → better IMM scores → improved gene discrimination, especially for small or atypical genomes.

**Comment by roeimed0** [comment]
Fixed in PR #102 (merged). Replaced the ad-hoc formula `floor(log2(n/min_obs)/2)` clamped to [3,8] with held-out log-likelihood selection via `_select_imm_order()`. The new method evaluates each candidate order on a held-out 20% of the intergenic set and picks the order that maximises log-likelihood, giving principled model selection without a hand-tuned formula.
